components/modal/repay-modal.tsx

```diff
@@ -38,47 +90,69 @@ const RepayModal = ({
-              <img
-                src={getLogo(position?.currency)}
-                alt={position?.currency.name}
-                className="w-5 h-5"
-              />
-              <div className="text-sm sm:text-base">
-                {position?.currency.symbol}
-              </div>
-            </div>
-            <div className="flex text-xs sm:text-sm gap-1 sm:gap-2">
-              <div className="text-gray-500">Available</div>
-              <div>{position?.amount}</div>
-              <button className="text-green-500">MAX</button>
-            </div>
-          </div>
-        </div>
+        <CurrencyAmountInput
+          currency={position.underlying}
+          value={value}
+          onValueChange={setValue}
+          price={prices[position.underlying.address]}
+          balance={min(
+            position.amount,
+            available,
+            balances[position.underlying.address],
+          )}
+        />{' '}
       </div>
       <div className="font-bold mb-3">Transaction Overview</div>
       <div className="flex flex-col gap-2 text-gray-500 text-sm mb-8">
-        <div>Remaining Debt</div>
-        <div>LTV</div>
+        <div className="flex gap-3 justify-between sm:justify-start">
+          <div className="text-gray-500">Remaining Debt</div>
+          <div>
+            {formatUnits(
+              position.amount,
+              position.underlying.decimals,
+              prices[position.underlying.address],
+            )}{' '}
+            {position.underlying.symbol}
+          </div>
+        </div>
+        <div className="flex gap-3 justify-between sm:justify-start">
+          <div className="text-gray-500">LTV</div>
+          <div className="flex items-center gap-1">
+            <span className="text-green-500">{position.ltv.toFixed(2)}%</span>
+            {value ? (
+              <>
+                <Arrow />
+                <span className="text-green-500">{ltv.toFixed(2)}%</span>
+              </>
+            ) : (
+              <></>
+            )}
+          </div>
+        </div>
       </div>
       <button
-        disabled={true}
-        className="font-bold text-base sm:text-xl disabled:bg-gray-100 dark:disabled:bg-gray-800 h-12 sm:h-16 rounded-lg disabled:text-gray-300 dark:disabled:text-gray-500"
+        disabled={
+          amount === 0n ||
+          amount > available ||
+          amount > position.amount ||
+          amount > balances[position.underlying.address]
+        }
+        className="font-bold text-base sm:text-xl bg-green-500 disabled:bg-gray-100 dark:disabled:bg-gray-800 h-12 sm:h-16 rounded-lg text-white disabled:text-gray-300 dark:disabled:text-gray-500"
+        onClick={async () => {
+          await repay(position, amount, refund)
+          setValue('')
+          onClose()
+        }}
       >
-        Confirm
+        {amount === 0n
+          ? 'Enter repay amount'
```

Comment:
Enter amount to repay

Reply:
https://github.com/clober-dex/coupon.finance/pull/94/commits/d575a957e1e75afac296582d67458acff4466016